### ISSUE TITLE
turf.intersect return null when one of arguments is single MultiPolygon

### ISSUE THREAD

**Issue by IMMSPgisgroup** [Turfjs/turf#1367]
I see some interesting in this simple sample.

```JavaScript
  var poly1 = {
    "type": "Feature",
    "geometry": {
      "type": "MultiPolygon",
      "coordinates": [[[
        [-122.801742, 45.48565],
        [-122.801742, 45.60491],
        [-122.584762, 45.60491],
        [-122.584762, 45.48565],
        [-122.801742, 45.48565]
      ]]]
    }
  };
  var poly2 = {
    "type": "Feature",
    "geometry": {
      "type": "MultiPolygon",
      "coordinates": [[[
        [-122.520217, 45.535693],
        [-122.64038, 45.553967],
        [-122.720031, 45.526554],
        [-122.669906, 45.507309],
        [-122.723464, 45.446643],
        [-122.532577, 45.408574],
        [-122.487258, 45.477466],
        [-122.520217, 45.535693]
      ]]]
    }
  }

  var intersection = turf.intersect(poly1, poly2); 
```

Similar MultiPolygons really is Polygons, but this is valid GeoJSON.
Such Features may return, for example, by query to PostGIS database.

In TurfJS 2.0.0 thats work fine and return Feature (intersection poly1 and poly 2).
In TurfJS 5.1.6 thats return null.

Is this regression?

**Comment by rowanwins** [comment]
Hi @IMMSPgisgroup 

Thanks for reporting, that is a bit weird. 

Under the hood both of both v2 and v5 `@turf/intersect` use JSTS which has been known to have problems with various multi-geometries, not sure why in this particular case things are running into problems and what would've changed since v2.

In v6 we're trying really hard to remove that dependency because it's been quite problematic and large. There is already a preliminary version published which you can try out, check out this [runkit example](https://runkit.com/rowanwins/turf-intersect). It works pretty much the same, just note that you need to require the default `function`.

Hope that helps.
Rowan

**Comment by IMMSPgisgroup** [comment]
Hi

Thanks for answer.
But we hove some truble.
We don't familiar with JavaScript tools.
We need for version, which may be use directly in browser 
Dear colleagues, give version, prepared for work in browser, fo us, please :(.
_______________________________________________________________________________
With respect, IMMSPgisgroup

**Comment by w8r** [comment]
@DenisCarriere, @rowanwins this one is fixed in martinez@0.4.x

**Comment by stephanietuerk** [comment]
I'm having this problem and I'm using turf v6.1.0. I'm not getting an error: I get null when I try to intersect a Polygon and MultiPolygon. 

Here are my polys: 
[https://gist.github.com/stephanietuerk/6374b83c88729002f26782d0e52e9c71](url)

code is simply:
`let intersection = turf.intersect(poly1, poly2));`

**Comment by rowanwins** [comment]
Resolved since v6.2.0
